### ISSUE TITLE
Possibility to add Nextcloud groups as attendees for an event

### ISSUE THREAD

**Issue by tcitworld** [nextcloud/calendar#2955]
Nextcloud groups as in user management, including LDAP.

To add Nextcloud contact groups as attendees for an event, see #167
To add a Nextcloud circle as attendees for an event, see #2954
To add a Nextcloud talk group as attendee for an event, see

**Comment by vimmerru** [comment]
@tcitworld Hi, is there any timeline for this feature? Is there any option to speedup this like a payment/donation?

**Comment by ChristophWurst** [comment]
There is no timeline right now, this is not a planned feature. you could try your luck at https://help.nextcloud.com/c/nextcloud-freelancing/48 to find someone who you can sponsor for this. Bountysource and other sponsoring platforms are not used at the moment.

Many of the maintainers work for Nextcloud Gmbh. If you are a customer there you can send a feature request and it will be taken in consideration for the roadmap plannings in the future.

**Comment by vimmerru** [comment]
@ChristophWurst will you accept the contribution? What acceptance criteria do you have?

**Comment by ChristophWurst** [comment]
Absolutely. I'd like to discuss the proposed solution first and not get just the full feature as a code dump to ensure the concept works and the code is in a structure that will be acceptable.

I would prefer if this feature is also unit tested, but I don't necessarily expect full coverage.

**Comment by maximelehericy** [comment]
Not sure where to drop this message, but talking with @ChristophWurst, a concern raised about the big challenge which would be keeping the attendees in sync with the group.
While I understand why keeping invite in sync with group could be useful, to me, a first iteration with only replacing a group by users at the moment the event is created would be fine, as it solve one of the main problems, being that when you have a group with a large number of users, you definitely don't want to invite them all one by one :)

**Comment by SebastianKrupinski** [comment]
Hi,

I think the best way to solve this would be to implement proper vCard groups.

https://github.com/nextcloud/contacts/issues/4009

Sebastian

**Comment by tcitworld** [comment]
You mean implementing support for #167 (contact groups) and using the system addressbook for NC groups?

**Comment by miaulalala** [comment]
> You mean implementing support for #167 (contact groups) and using the system addressbook for NC groups?

I think it's more only supporting VCARD v4 and allowing proper VCARD groups

**Comment by SebastianKrupinski** [comment]
> > You mean implementing support for #167 (contact groups) and using the system addressbook for NC groups?
> 
> I think it's more only supporting VCARD v4 and allowing proper VCARD groups

Yes, the standard allows for all kinds of group types with custom URNs so the groups could be used for everything from Email groups to instant messaging groups and even internal NC groups.

**Comment by small1** [comment]
ping on this one. Any updates?

**Comment by small1** [comment]
Any updates on this?  As the web interface states groups as attendees it is quite confusing for end users not to get any results for it.

**Comment by okworx** [comment]
Here's a screen shot where groups are specifically mentioned in the search field for attendees:

![Image](https://github.com/user-attachments/assets/7738be5d-924f-4bb4-b6f0-3265ee0bd5d7)

**Comment by ChristophWurst** [comment]
@okworx that is because of https://github.com/nextcloud/calendar/issues/167.

**Comment by small1** [comment]
The formulation is confusing since it states groups. And users take it as user groups not contact groups. They get confused :)  Since groups everywhere else in Nextcloud is user groups.

**Comment by ChristophWurst** [comment]
Ambiguous UI string adjusted with https://github.com/nextcloud/calendar/pull/6810

**Comment by Abekonge** [comment]
A workaround is to use the Contacts app to make a team, based on a Nextcould group. Then you can add the team to an invitation, and it will be the Nextcloud group...

**Comment by mirenbz** [comment]
Thanks for the suggestion, @Abekonge. Note that "contact groups" will not take system users, so it needs to be a "Team". Works perfectly to add them to calendar events though 👌🏼
